pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,291 @@
+	EmptyResourceList = resourcecondition.ResourceList{
+		APIVersion: "v1",
+		Items:      []interface{}{},
+		Kind:       "List",
+		Metadata: struct {
+			ResourceVersion string `json:"resourceVersion"`
+		}{
+			ResourceVersion: "",
+		},
+	}
+)
+
+// kubectl return the kubectl command
+// If the environment variable COMMAND is set, it will return the value of COMMAND
+// Otherwise, it will return the default value "kubectl" as default
+// Arguments:
+// - format: format of the command without kubeclt or oc
+// - args: arguments of the command
+func kubectl(format string, args ...interface{}) string {
+	binary := DefaultCommandTool
+	if cmd := os.Getenv("COMMAND"); cmd != "" {
+		binary = cmd
+	}
+
+	return fmt.Sprintf("%s "+format, append([]interface{}{binary}, args...)...)
+}
+
+// ApplyString applies the given yaml string to the cluster
+func ApplyString(yamlString string) error {
+	cmd := kubectl("apply --server-side -f -")
+	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		return fmt.Errorf("error applying yaml: %v", err)
+	}
+
+	return nil
+}
+
+// DeleteString delete the given yaml string to the cluster
+func DeleteString(yamlString string) error {
+	cmd := kubectl("delete -f -")
+	output, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		// Workaround because the resource can be already deleted
+		if strings.Contains(output, "not found") {
+			return nil
+		}
+
+		return fmt.Errorf("error deleting yaml: %v", err)
+
+	}
+
+	return nil
+}
+
+// GetResourceCondition returns the condition of a resource
+func GetResourceCondition(ns, resourceType, resourceName string) ([]resourcecondition.Condition, error) {
+	var resource resourcecondition.Resource
+
+	output, err := GetResource(ns, resourceType, resourceName)
+	if err != nil {
+		return []resourcecondition.Condition{}, err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return []resourcecondition.Condition{}, err
+	}
+
+	return resource.Status.Conditions, nil
+}
+
+// GetPodPhase returns the phase of a pod
+func GetPodPhase(ns, selector string) (string, error) {
+	var resource resourcecondition.Resource
+
+	podName, err := GetPodFromLabel(ns, selector)
+	if err != nil {
+		return "", err
+	}
+
+	output, err := GetResource(ns, "pod", podName)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return "", err
+	}
+
+	return resource.Status.Phase, nil
+}
+
+// GetCRDs returns all the CRDs names in a list
+func GetCRDs() ([]string, error) {
+	cmd := kubectl("get crds -o jsonpath={.items[*].metadata.name}")
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return []string{}, fmt.Errorf("error getting crds: %v", err)
+	}
+
+	return strings.Split(output, " "), nil
+}
+
+// GetAllResources returns all the resources of a namespace
+func GetAllResources(ns string) (resourcecondition.ResourceList, error) {
```

Comment:
This returns JSON, right? So it should state this in the function name, otherwise we're inconsistent (we have other functions that have similar names, but they only return the name).

Reply:
Yes returns a JSON but it's really a resourceList, so I'll rename to GetResourceList